code/modules/power/lighting.dm

```diff
@@ -745,8 +745,7 @@
 	if(zap_flags & ZAP_MACHINE_EXPLOSIVE)
 		explosion(src,0,0,0,flame_range = 5, adminlog = FALSE)
 		qdel(src)
-	else
-		return ..()
+	. = ..(power, zap_flags & ~ZAP_MACHINE_EXPLOSIVE & ~ZAP_OBJ_DAMAGE)
```

Comment:
Calling parent after deleting is a bad idea.
Also same as the other ones, don't modify the arguments in the parent call, do it in the proc properly.

Reply:
ahhh gottcha